contrib/hbcurl/core.c

```diff
@@ -1034,6 +1051,50 @@ HB_FUNC( CURL_EASY_SETOPT )
                }
             }
             break;
+            case HB_CURLOPT_HTTPPOST_FORM:
+            {
+               PHB_ITEM pArray = hb_param( 3, HB_IT_ARRAY );
+
+               if( pArray )
+               {
+                  HB_SIZE ulPos;
+                  HB_SIZE ulArrayLen = hb_arrayLen( pArray );
+
+                  for( ulPos = 0; ulPos < ulArrayLen; ++ulPos )
+                  {
+                     PHB_ITEM pSubArray = hb_arrayGetItemPtr( pArray, ulPos + 1 );
+
+                     int type = hb_arrayGetNI( pSubArray, 1 );
+
+                     switch( type )
+                     {
+                        case 1:
+
+                           curl_formadd( &hb_curl->pHTTPPOST_First,
+                                         &hb_curl->pHTTPPOST_Last,
+                                         CURLFORM_COPYNAME, hb_arrayGetCPtr( pSubArray, 2 ),
+                                         CURLFORM_NAMELENGTH, hb_arrayGetCLen( pSubArray, 2 ),
+                                         CURLFORM_COPYCONTENTS, hb_arrayGetCPtr( pSubArray, 3 ),
+                                         CURLFORM_CONTENTSLENGTH, hb_arrayGetCLen( pSubArray, 3 ),
+                                         CURLFORM_END );
+
+                           break;
+                        case 2:
```

Comment:
UPDATE: confused two issues here, the FILE issue with the notion that it'd be better to use Harbour file I/O instead of relying on curl's own functions. I'd tend to drop the `CURLOPT_STDERR` thing and focus on creating a file reader with CURLFORM_STREAM.